test/debug/reflection_test.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'test/unit'
+require_relative '../../lib/debug/reflection'
+
+module DEBUGGER__
+  class ReflectionTest < Test::Unit::TestCase
+    def setup
+      @sample_object = SampleClass.new(1, 2)
+    end
+
+    def test_instance_variables_of
+      assert_equal [:@a, :@b], Reflection.instance_variables_of(@sample_object)
+    end
+
+    def test_instance_variables_get
+      assert_equal 1, Reflection.instance_variable_get_from(@sample_object, :@a)
+      assert_equal 2, Reflection.instance_variable_get_from(@sample_object, :@b)
+    end
+
+    def test_class_of
+      assert_same SampleClass, Reflection.class_of(@sample_object)
+    end
+
+    def test_singleton_class_of
+      expected = class << SampleClass
+        self
+      end
+
+      assert_same expected, Reflection.singleton_class_of(SampleClass)
+    end
+
+    def test_is_kind_of?()
+      assert_true Reflection.is_kind_of?(@sample_object, SampleClass)
+      assert_false Reflection.is_kind_of?(@sample_object, Object)
+    end
+
+    def test_responds_to?
+      assert_true Reflection.responds_to?(@sample_object, :a)
+      assert_false Reflection.responds_to?(@sample_object, :doesnt_exist)
+
+      assert_false Reflection.responds_to?(@sample_object, :sample_private_method)
+      assert_false Reflection.responds_to?(@sample_object, :sample_private_method, include_all: false)
+      assert_true Reflection.responds_to?(@sample_object, :sample_private_method, include_all: true)
+    end
+
+    def test_method_of
+      assert_equal 1, Reflection.method_of(@sample_object, :a).call
+    end
+
+    def test_object_id_of
+      assert_equal @sample_object.__id__, Reflection.object_id_of(@sample_object)
+    end
+
+    def test_name_of
+      assert_equal "DEBUGGER__::ReflectionTest::SampleClass", Reflection.name_of(SampleClass)
+    end
+
+    def test_bind_call_backport
+      omit_if(
+        UnboundMethod.instance_method(:bind_call).source_location.nil?,
+        "This Ruby version (#{RUBY_VERSION}) has a native #bind_call implementation, so it doesn't need the backport.",
+      )
```

Comment:
This causes a fair bit of noise in the test output:

```
=================================================================================================================================================
Omission: This Ruby version (3.2.2) has a native #bind_call implementation, so it doesn't need the backport. [test_bind_call_backport(DEBUGGER__::ReflectionTest)]
./test/debug/reflection_test.rb:60:in `test_bind_call_backport'
=================================================================================================================================================
Finished in 0.001056 seconds.
-------------------------------------------------------------------------------------------------------------------------------------------------
10 tests, 15 assertions, 0 failures, 0 errors, 0 pendings, 1 omissions, 0 notifications
```

Perhaps it would be better to just wrap the `def test_bind_call_backport` in a big `if` statement?